pkg/scheduling/volumelimits.go

```diff
@@ -0,0 +1,202 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduling
+
+import (
+	"context"
+	"fmt"
+
+	"knative.dev/pkg/logging"
+
+	v1 "k8s.io/api/core/v1"
+	sv1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// VolumeLimits tracks volume limits on a per node basis.  The number of volumes that can be mounted varies by instance
+// type. We need to be aware and track the mounted volume usage to inform our awareness of which pods can schedule to
+// which nodes.
+type VolumeLimits struct {
+	ctx        context.Context
+	volumes    volumeUsage
+	podVolumes map[types.NamespacedName]volumeUsage
+	kubeClient client.Client
+}
+
+type volumeUsage map[string]sets.String
+
+func (u *volumeUsage) Add(provisioner string, pvcID string) {
```

Comment:
IIUC, since you type is itself a pointer (i.e. map), you don't need to use pointer receivers.
```suggestion
func (u volumeUsage) Add(provisioner string, pvcID string) {
```